File: 2022/day03.R
rucksacks <- strsplit(readLines("input03.txt"),"")
priority <- function(x)which(c(letters,LETTERS)==x)
sum(sapply(rucksacks, function(x){
  M <- matrix(x,ncol=2,byrow=FALSE)
  priority(unique(M[M[,1]%in%M[,2],1]))
  }))
#8039

n <- rep(0,length(rucksacks))
for(i in seq(1,length(rucksacks),3)){
  u <- unique(rucksacks[[i]][rucksacks[[i]]%in%rucksacks[[i+1]]])
  n[i] <- priority(unique(u[u%in%rucksacks[[i+2]]]))
}
sum(n)
#2510

File: 2022/other_languages/day03.jl
s = open("input03.txt") do file
    readlines(file)
end

rucksacks = collect.(s)

n=0
for i in 1:lastindex(rucksacks)
    z = [k for k in rucksacks[i] if k in rucksacks[i][1:end÷2] && k in rucksacks[i][end÷2+1:end]][1]
    if islowercase(z)
        n += z-'a'+1
    else
        n += z-'A'+27
    end
end

n
#8039

n=0
for i in 1:3:lastindex(rucksacks)
    z = [k for k in rucksacks[i] if k in rucksacks[i+1] && k in rucksacks[i+2]][1]
    if islowercase(z)
        n += z-'a'+1
    else
        n += z-'A'+27
    end
end

n
#2510